azure_li_services/units/install.py

```diff
@@ -43,7 +45,11 @@ def main():
         location='/mnt', label='azconfig'
     )
 
-    if 'directory' in packages_config:
```

Comment:
I'm not quite square here. The intent of what I'm after:

1. If there is no `packages:` section at all, that should not be considered an error. Exit out. I don't see that around line 39, but should probably be there.

1. If there is a `packages:` section, then any invalid data/missing fields from that point forward is an error. By the same token, an invalid directory specification (no such directory or whatever) is an error as well.

Please make sure that the code reflects that properly, thanks.

Reply:
man you are so right. That was too short thinking from my side. Don't approve when I'm writing crap :-) fix on its way

Reply:
Note that this is true of ALL services.

If there is no CPU or MEMORY tag, don't check that. If there is no Network tag, don't configure network (>>> shudder <<<, someone would only do that if they **loved** the KVM). If there is no Storage tag, don't configure disks. You get the idea.

Longer term, `credentials` is special. Once temporary password is gone, then lack of `credentials` means you won't be able to log in. That should be an error, because you can't actually log into the system! 😢

Reply:
sure all other services we have so far respects this idea.

The constraint checker service for cpu and men is on my list and does not yet exist. Of course it will not run the service if no machine_constraints section exists

regarding credentials we need another issue to raise an error if no credentials are specified. Could you please open that one ? Thanks